src/main/java/jenkins/branch/BranchStatusColumn.java

```diff
@@ -104,4 +106,19 @@ public boolean shownByDefault() {
             return false;
         }
     }
+
+    /**
+     * Hide this column from user views as it would only confuse them.
+     */
+    @Extension
+    public static class DescriptorVisibilityFilterImpl extends DescriptorVisibilityFilter {
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public boolean filter(Object context, Descriptor descriptor) {
+            return !(descriptor instanceof DescriptorImpl) || context instanceof MultiBranchProjectViewHolder.ViewImpl;
```

Comment:
won't `context instanceof MultiBranchProjectViewHolder.ViewImpl` hide all descriptors if the view is of that type?